cli/web_worker.rs

```diff
@@ -3,17 +3,59 @@ use crate::ops;
 use crate::state::State;
 use crate::worker::Worker;
 use crate::worker::WorkerEvent;
+use crate::worker::WorkerHandle;
+use deno_core::v8;
 use deno_core::ErrBox;
 use deno_core::StartupData;
+use futures::channel::mpsc;
 use futures::future::FutureExt;
 use futures::stream::StreamExt;
-use futures::SinkExt;
 use std::future::Future;
 use std::ops::Deref;
 use std::ops::DerefMut;
 use std::pin::Pin;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
 use std::task::Context;
 use std::task::Poll;
+
+#[derive(Clone)]
+pub struct WebWorkerHandle {
+  worker_handle: WorkerHandle,
+  terminate_tx: mpsc::Sender<()>,
+  terminated: Arc<AtomicBool>,
+  isolate_handle: v8::IsolateHandle,
+}
+
+impl Deref for WebWorkerHandle {
+  type Target = WorkerHandle;
+  fn deref(&self) -> &Self::Target {
+    &self.worker_handle
+  }
+}
+
+impl DerefMut for WebWorkerHandle {
+  fn deref_mut(&mut self) -> &mut Self::Target {
+    &mut self.worker_handle
+  }
+}
+
+impl WebWorkerHandle {
+  pub fn terminate(&self) {
+    let already_terminated = self.terminated.swap(true, Ordering::Relaxed);
+
+    if !already_terminated {
+      self.isolate_handle.terminate_execution();
+      let mut sender = self.terminate_tx.clone();
+      sender
+        .try_send(())
+        .map_err(ErrBox::from)
+        .expect("Failed to terminate");
```

Comment:
`.map_err(ErrBox::from)` is unnecessary.

Reply:
Consider adding some comments here describing the situation in some detail. We are okay with the send failing, correct?